tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/PublicAssetSetMethod.scala

```diff
@@ -127,4 +131,35 @@ class PublicAssetSetCreatePerformer @Inject()(val publicAssetRepository: PublicA
         case _: BlobNotFoundException => PublicAssetBlobIdNotFoundException(creationRequest.blobId.value.value)
         case _: IllegalArgumentException => PublicAssetInvalidBlobIdException(creationRequest.blobId.value.value)
       }
+}
+
+
+class PublicAssetSetUpdatePerformer @Inject()(val publicAssetRepository: PublicAssetRepository,
+                                              val publicAssetSetService: PublicAssetSetService) {
+
+  def updatePublicAssets(mailboxSession: MailboxSession, request: PublicAssetSetRequest): SMono[PublicAssetUpdateResults] =
+    tryGetValidateUpdateRequest(request)
+      .flatMap({
+        case Left(updateFailure) => SMono.just(updateFailure)
+        case Right((publicAssetId, validatedPath)) => updatePublicAsset(publicAssetId, validatedPath, mailboxSession)
+      }, ReactorUtils.DEFAULT_CONCURRENCY)
+      .collectSeq()
+      .map(PublicAssetUpdateResults)
+
+  private def updatePublicAsset(publicAssetId: PublicAssetId, patch: ValidatedPublicAssetPatchObject, mailboxSession: MailboxSession): SMono[PublicAssetUpdateResult] =
+    publicAssetSetService.checkIdentityIdsExist(patch.identityIds.toSeq, mailboxSession)
+      .flatMap(checkedIdentityIds => SMono(publicAssetRepository.update(mailboxSession.getUser, publicAssetId, checkedIdentityIds.toSet)))
+      .`then`(SMono.just[PublicAssetUpdateResult](PublicAssetUpdateSuccess(publicAssetId)))
+      .onErrorResume(e => SMono.just(PublicAssetUpdateFailure(UnparsedPublicAssetId(publicAssetId.value.toString), e)))
+
+  private def tryGetValidateUpdateRequest(request: PublicAssetSetRequest): SFlux[Either[PublicAssetUpdateFailure, (PublicAssetId, ValidatedPublicAssetPatchObject)]] =
+    SFlux.fromIterable(request.update.getOrElse(Map.empty))
+      .map({
+        case (unparsedPublicAssetId: UnparsedPublicAssetId, patch: PublicAssetPatchObject) =>
+          (for {
+            publicAssetId <- unparsedPublicAssetId.tryAsPublicAssetId
+            validatedPath <- patch.validate()
+          } yield (publicAssetId, validatedPath))
+            .left.map(e => PublicAssetUpdateFailure(unparsedPublicAssetId, e))
```

Comment:
If I specify an invalid JSON, do I have a server log about it?

Reply:
log at PublicAssetUpdateFailure.asSetError